File: 509. Fibonacci Number/509.FibonacciNumber-iterative.js
var fib = function (targetN) {
  let currentN = 1;
  let a = 0,
    b = 1;

  if (targetN === 0) return 0;

  for (let i = 2; i <= targetN; i++) {
    let temp = a + b;
    a = b;
    b = temp;
  }

  return b;
};

File: 509. Fibonacci Number/509.FibonacciNumber-iterative.java
public class Solution {
    public int fib(int targetN) {
        if (targetN == 0) return 0;
        if (targetN == 1) return 1;

        int a = 0, b = 1;
        for (int i = 2; i <= targetN; i++) {
            int temp = a + b;
            a = b;
            b = temp;
        }
        return b;
    }
}